tests/test_utilities.py

```diff
@@ -296,3 +296,24 @@ def test_avoid_repetitions():
     assert 3 * len(result.year) == 2 * len(assets.year)
     original = result.interp(year=assets.year, method="linear")
     assert (original == assets).all()
+
+
+def test_check_dimensions():
+    from muse.utilities import check_dimensions
+
+    data = xr.DataArray(
+        np.random.rand(4, 5),
+        dims=["dim1", "dim2"],
+        coords={"dim1": range(4), "dim2": range(5)},
+    )
+
+    # Valid
+    check_dimensions(data, required=["dim1"], optional=["dim2"])
+
+    # Missing required
+    with raises(ValueError):
```

Comment:
Check the error text as well to make sure it fails for the right reason:
```suggestion
    with raises(ValueError, match="Missing required dimensions"):
```